### PR TITLE
Add file dropdown menu action plugins framework

### PR DESCRIPTION
**Pull request by chetanyakan**

### CODE REVIEW

File: components/file_attachment/file_attachment.tsx [hunk 1]
```diff
@@ -121,6 +139,122 @@ export default class FileAttachment extends React.PureComponent<Props, State> {
...
+                <Menu.ItemAction
+                    id={item.id + '_pluginmenuitem'}
+                    key={item.id + '_pluginmenuitem'}
+                    onClick={() => item?.action(fileInfo)}
+                    text={item.text}
+                />
+            );
+        });
+
+        if (!defaultItems?.length && !pluginItems?.length) {
```
Comment (hrishavkmr) [inline components/file_attachment/file_attachment.tsx]: condition can be simplified
```suggestion
        if (!(defaultItems?.length || pluginItems?.length)) {
```

**Review by hrishavkmr — APPROVED** [review]

File: components/search_results/search_results.tsx [hunk 2]
```diff
@@ -73,6 +77,7 @@ const defaultProps: Partial<Props> = {
 const SearchResults: React.FC<Props> = (props: Props): JSX.Element => {
     const scrollbars = useRef<Scrollbars|null>(null);
     const [searchType, setSearchType] = useState<string>(props.searchType);
+    const filesDropdownPluginMenuItems = useSelector((state: GlobalState) => state.plugins.components.FilesDropdown || []) as unknown as FileDropdownPluginComponent[];
```
Comment (manojmalik20) [inline components/search_results/search_results.tsx:80]: We can make a selector for this as it may be used in multiple files.

**Review by manojmalik20 — APPROVED** [review]

File: components/file_attachment/file_attachment.tsx (hunk 1 above)
Comment (chetanyakan) [reply components/file_attachment/file_attachment.tsx]: done

**Comment by chetanyakan** [comment]
Closing as created a PR in the mattermost org.

**Pull request closed without merging**